src/reporter.js

```diff
@@ -1,15 +1,16 @@
 const SauceLabs = require('saucelabs').default;
 const path = require('path');
 const fs = require('fs');
-const crypto = require('crypto');
-const { tmpdir } = require('os');
-const { promisify } = require('util');
-
-const writeFile = promisify(fs.writeFile);
-const rmdir = promisify(fs.rmdir);
+const { readFile } = require('fs/promises');
 
 class Reporter {
   constructor (cypressDetails) {
+    let reporterVersion = 'unknown';
+    try {
+      const packageData = JSON.parse(fs.readFileSync(path.join(__dirname, '..', 'package.json')));
```

Comment:
A bit more hardened and shows the intent of wanting to parse a string.

```suggestion
      const packageData = JSON.parse(fs.readFileSync(path.join(__dirname, '..', 'package.json'), 'utf-8'));
```